### ISSUE TITLE
bug: capacity overflow in repeat function

### ISSUE THREAD

**Issue by andygrove** [apache/datafusion-comet#482]
### Describe the bug

Query works in Spark, fails in Comet. I discovered this bug using [Comet Fuzz](https://github.com/apache/datafusion-comet/pull/472).
```
SELECT repeat(c8, c5) FROM test1
```

```
Query failed in Comet: Job aborted due to stage failure: Task 0 in stage 73.0 failed 1 times, most recent failure: Lost task 0.0 in stage 73.0 (TID 546) (192.168.86.33 executor driver): org.apache.comet.CometNativeException: capacity overflow
```

**Comment by tshauck** [comment]
I think the root cause of this is that for negative numbers, datafusion's repeat scalar function panics with `capacity overflow`. I think https://github.com/apache/datafusion/pull/10760 should fix that upstream, which hopefully will fix this once it makes it into datafusion-comet.

**Comment by andygrove** [comment]
Thanks @tshauck. It is nice to see the fuzzing resulting in improvements upstream. This does seem like an edge case that users are unlikely to hit so I think we can wait until we upgrade to DataFusion 39 and then retest and close this issue.

**Comment by tshauck** [comment]
@andygrove I tried running Comet Fuzz now that DataFusion is at version 39 for Comet so it includes https://github.com/apache/datafusion/pull/10760, and it looks like this issue may be resolved (or at least it didn't show up in my 
[results-1718078990598.md](https://github.com/user-attachments/files/15782084/results-1718078990598.md) file). I also manually ran `repeat` via spark-sql and it doesn't panic any more on negative numbers.

Obviously, your call to close, but just wanted to follow up given update to DataFusion core.
